world.js

```diff
@@ -294,7 +295,16 @@ const _bindHitTracker = app => {
   hitTracker.bind(app);
 
   const die = () => {
```

Comment:
I think this is backwards. What controls the dying should be the NPC. The app should not be aware of NPCs. The NPC controls the app.